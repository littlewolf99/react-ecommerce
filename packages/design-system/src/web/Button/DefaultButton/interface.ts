import { TColorsTypes } from '../../../interfaces';

export type TDefaultButtonProps = {
  text: string;
  onClick: () => void;
  bg: TColorsTypes;
}

export type TContainerStyleProps = {
  width: number;
  height: number;
  bg: TColorsTypes;
}
