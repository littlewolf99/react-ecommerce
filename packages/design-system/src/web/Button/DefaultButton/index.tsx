import React, { FunctionComponent } from 'react';

import Container from './styles';

import { TDefaultButtonProps, TGetVariants } from './interface';

const Button: FunctionComponent<TDefaultButtonProps> = ({ text, onClick, variant }) => {
  const getButtonVariants = () => {
    const getVariants: TGetVariants = {
      primary: {
        bg: 'primary',
        hoverColor: 'primaryDark',
      },
      secondary: {
        bg: 'secondary',
        hoverColor: 'secondaryDark',
      },
    };

    return getVariants[variant] || getVariants.primary;
  };

  return (
    <Container
      width={150}
      height={32}
      onClick={onClick}
      bg={getButtonVariants().bg}
      hoverColor={getButtonVariants().hoverColor}
      border="none"
    >
      {text}
    </Container>
  );
};

export default Button;
