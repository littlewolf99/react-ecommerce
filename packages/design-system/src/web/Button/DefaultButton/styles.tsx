import styled from 'styled-components';
import { color, layout, border } from 'styled-system';

import { TContainerStyleProps } from './interface';

const Container = styled.button<TContainerStyleProps>`
  ${color}
  ${layout}
  ${border}
`;

export default Container;
