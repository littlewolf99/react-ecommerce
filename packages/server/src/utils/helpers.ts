type GenerateSkuProps = {
  name: string;
  colorName: string;
  brand: string;
};

export class Helpers {
  generateSku({ name, colorName, brand }: GenerateSkuProps) {
    const nameSplitted = name.split(' ');

    const nameCode =
      nameSplitted.length > 1
        ? `${nameSplitted[0].charAt(0)}${nameSplitted[1].charAt(0)}`
        : `${nameSplitted[0].charAt(0)}${nameSplitted[0].charAt(1)}`;

    const colorCode = colorName.substr(0, 2); //TODO - Create a new columns at colors table that have an color code

    const brandCode = brand.substr(0, 3);

    const sku = `${nameCode}-${colorCode}-${brandCode}`;

    return sku.toUpperCase();
  }
}
