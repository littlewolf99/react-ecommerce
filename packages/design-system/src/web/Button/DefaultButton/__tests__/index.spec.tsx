import React from 'react';
import { mountWithTheme } from '../../../../../__tests__/helpers';

import DefaultButton from '..';

describe('Unit | DefaultButton', () => {
  const onPress = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should create snapshot', () => {
    const wrapper = mountWithTheme(
      <DefaultButton onPress={onPress} text="Press me" variant="primary" />,
    );

    expect(wrapper.text()).toMatchSnapshot();
  });

  it('should render a Button with a label Press me', () => {
    const wrapper = mountWithTheme(
      <DefaultButton onPress={onPress} text="Press me" variant="primary" />,
    );

    expect(wrapper.text()).toContain('Press me');
  });

  it('should call the onPress function one time', () => {
    const wrapper = mountWithTheme(
      <DefaultButton testID="button" onPress={onPress} text="Press me" variant="primary" />,
    );

    const onPressHandler = wrapper
      .find('[testID="button"]')
      .first()
      .prop('onPress') as () => void;

    onPressHandler();

    expect(onPress).toBeCalledTimes(1);

    wrapper.unmount();
  });

  it.skip('should render a Button with a Loading', () => {
    const wrapper = mountWithTheme(
      <DefaultButton loading onPress={onPress} text="Press me" />,
    );

    expect(wrapper.find('[testID="loadingIndicator"]').exists()).toBeTruthy();
  });

  it.skip('should call the onPress function zero times if the button is loading', () => {
    const wrapper = mountWithTheme(
      <DefaultButton testID="button" loading onPress={onPress} text="Press me" />,
    );

    const onPressHandler = wrapper
      .find('[testID="button"]')
      .first()
      .prop('onPress') as () => void;

    onPressHandler();

    expect(onPress).toBeCalledTimes(0);
  });

  it('should compare a snapshot with a disabled style', () => {
    const wrapper = mountWithTheme(
      <DefaultButton disabled onPress={onPress} text="Press me" />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should compare a snapshot with a loading style', () => {
    const wrapper = mountWithTheme(
      <Button loading onPress={onPress} text="Press me" />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
