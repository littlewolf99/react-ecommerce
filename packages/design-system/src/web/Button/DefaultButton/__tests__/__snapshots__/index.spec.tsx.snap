// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit | DefaultButton should compare a snapshot with a disabled style 1`] = `
<Button
  disabled={true}
  onPress={[MockFunction]}
  text="Press me"
>
  <styled.button
    bg="gray"
    border="none"
    disabled={true}
    height={32}
    hoverColor="gray"
    width={150}
  >
    <StyledComponent
      bg="gray"
      border="none"
      disabled={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "gxfjJc",
            "rules": Array [
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "

  box-shadow: 0px 10px 12px rgba(0,0,0,0.2);

  &:hover {
    background-color: ",
              [Function],
              "
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      height={32}
      hoverColor="gray"
      width={150}
    >
      <button
        className="sc-bdVaJa gxfjJc"
        disabled={true}
        height={32}
        width={150}
      >
        Press me
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;

exports[`Unit | DefaultButton should create snapshot 1`] = `"Press me"`;
